tests/integration/acp/relationship/add_doc_actor_test/add_doc_actor_with_manager_test.go

```diff
@@ -0,0 +1,1286 @@
+				Identity: immutable.Some(1),
+
+				CollectionID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad",
+						"age": 28
+					}
+				`,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(2), // This identity (to be manager) can not read yet.
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{}, // Can't see the documents yet
+				},
+			},
+
+			testUtils.UpdateDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(2), // Manager can't update yet.
+
+				DocID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad Lone"
+					}
+				`,
+
+				ExpectedError: "document not found or not authorized to access",
+			},
+
+			testUtils.DeleteDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(2), // Manager can't delete yet.
+
+				DocID: 0,
+
+				ExpectedError: "document not found or not authorized to access",
+			},
+
+			testUtils.AddDocActorRelationship{ // Make admin / manager
+				RequestorIdentity: 1,
+
+				TargetIdentity: 2,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "admin",
+
+				ExpectedExistence: false,
+			},
+
+			testUtils.AddDocActorRelationship{ // Manager makes itself a writer
+				RequestorIdentity: 2,
+
+				TargetIdentity: 2,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "writer",
+
+				ExpectedExistence: false,
+			},
+
+			// Note: It is not neccesary to make itself a reader, as becoming a writer allows reading.
+			testUtils.AddDocActorRelationship{ // Manager makes itself a reader
+				RequestorIdentity: 2,
+
+				TargetIdentity: 2,
+
+				CollectionID: 0,
+
+				DocID: 0,
+
+				Relation: "reader",
+
+				ExpectedExistence: false,
+			},
+
+			testUtils.UpdateDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(2), // Manager can now update.
+
+				DocID: 0,
+
+				Doc: `
+					{
+						"name": "Shahzad Lone"
+					}
+				`,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(2), // Manager can read now
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{
+						{
+							"_docID": "bae-9d443d0c-52f6-568b-8f74-e8ff0825697b",
+							"name":   "Shahzad Lone",
+							"age":    int64(28),
+						},
+					},
+				},
+			},
+
+			testUtils.DeleteDoc{
+				CollectionID: 0,
+
+				Identity: immutable.Some(2), // Manager can now delete.
+
+				DocID: 0,
+			},
+
+			testUtils.Request{
+				Identity: immutable.Some(2), // Make sure manager was able to delete the document.
+
+				Request: `
+					query {
+						Users {
+							_docID
+							name
+							age
+						}
+					}
+				`,
+
+				Results: map[string]any{
+					"Users": []map[string]any{},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestACP_ManagerCanNotGivesAccessNotDefinedInPolicyToAnotherActor_Error(t *testing.T) {
```

Comment:
suggestion: I think it should be "TestACP_ManagerGivesAccessNotDefinedInPolicyToAnotherActor_Error"

Reply:
- [x] Changed to a better name: `TestACP_ManagerAddsRelationshipWithRelationItDoesNotManageAccordingToPolicy_Error`